src/worker/dispatch_data.rs

```diff
@@ -37,28 +37,18 @@ impl ClipboardDispatchData {
 
     /// Set the last observed seat.
     pub fn set_last_observed_seat(&mut self, seat: WlSeat, serial: u32) {
-        let pos = self.observed_seats.iter().position(|st| st.0 == seat);
-        let (seat, serial) = match pos {
-            Some(pos) => {
-                // We just found that `pos` we're going to remove, so unwrapping is safe.
-                self.observed_seats.remove(pos).unwrap()
-            }
-            None => (seat, serial),
-        };
```

Comment:
This should be the problematic code. Getting the `serial` here from the removed element rather than using the serial passed into the function leads to a reuse of the old serial updating absolutely nothing.